### ISSUE TITLE
Nixpkgs manual scope

### ISSUE THREAD

**Issue by lheckemann** [NixOS/nixpkgs#26211]
The nixpkgs manual is entitled "Nixpkgs Contributors Guide". Yet it contains various sections that are very relevant for people who intend to use nixpkgs without making any changes to it:

 - Language-specific development tools; I think in this case it makes sense, as development will usually be integrated quite strongly with nixpkgs when nix is being used for development, and packaging an application for nixpkgs and setting up a nix-based development environment are very similar tasks.

- Setup documentation for end-user applications. These are scattered somewhat inconsistently around the manual. The most consistent location for these is [the package notes section](https://nixos.org/nixpkgs/manual/#chap-package-notes) — but this includes (AFAIU) maintainer-oriented info in the elm subsection too. In addition, some docs ([vim config](https://nixos.org/nixpkgs/manual/#users-guide-to-vim-pluginsaddonsbundlesscripts-in-nixpkgs), [TeXLive](https://nixos.org/nixpkgs/manual/#sec-language-texlive)) that fall under this category do not live in the Package Notes section.

- User documentation for nixpkgs as a whole, like [the configuration section](https://nixos.org/nixpkgs/manual/#chap-packageconfig).

 - Use of nixpkgs for packaging that is not intended as a contribution to nixpkgs. The [Overlays section](https://nixos.org/nixpkgs/manual/#chap-overlays) is the only one that targets this use case specifically, and I think it also makes sense for this to live here as it's related to the machinery in nixpkgs.

I've touched on this in some other issues previously (https://github.com/NixOS/nixpkgs/issues/22599#issuecomment-297859902, https://github.com/NixOS/nixpkgs/pull/25274#issue-224850963), but I thought it would be a good idea to have an issue to discuss this problem specifically.

tl;dr: I think there's not enough distinction between documentation for people who want to *use* nixpkgs vs people who want to *build on* nixpkgs.

**Comment by FRidh** [comment]
I think everybody agrees the Nixpkgs guide is a mixture of things and that it would be good to have a distinct manual for users. The question is what to include there, considering the overlap.

Last year I sent the following mail to the list. 
http://www.mail-archive.com/nix-dev@lists.science.uu.nl/msg18125.html
There have been many more discussions regarding docs.

**Comment by mmahut** [comment]
Are there any updates on this issue, please

**Comment by lheckemann** [comment]
Still not clearly defined.

**Comment by FRidh** [comment]
The Nixpkgs manual is in the process of being migrated to CommonMark. As soon as that is done, and we have a different toolchain for producing the manual, we can continue this part.
